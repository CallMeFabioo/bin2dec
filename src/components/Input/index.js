import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const Input = ({
	id,
	placeholder,
	hasError,
	onChange = (_) => _,
	...props
}) => (
	<input
		className={classnames({
			border: true,
			rounded: true,
			shadow: true,
			'border-red-500': hasError,
			'w-full': true,
			'px-3': true,
			'py-2': true,
			'leading-tight': true,
			'text-gray-700': true,
			'appearance-none': true,
			'focus:outline-none': true,
			'focus:shadow-outline': true,
		})}
		type="text"
		maxLength="8"
		autoComplete="off"
		id={id}
		placeholder={placeholder}
		onChange={onChange}
		{...props}
	/>
);

Input.displayName = 'Input';

Input.propTypes = {
	id: PropTypes.string,
	placeholder: PropTypes.string,
	hasError: PropTypes.bool,
	onChange: PropTypes.func,
};

export default Input;
