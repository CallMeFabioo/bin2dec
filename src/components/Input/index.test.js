import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Input from '.';

it('renders a input', () => {
	const { container } = render(<Input placeholder="Input" />);

	expect(container).toMatchSnapshot();
});

it('renders a input with error status', () => {
	render(<Input placeholder="Input" hasError={true} />);

	expect(screen.getByRole('textbox')).toHaveClass('border-red-500');
});

it('triggers a onChange prop when value changes', async () => {
	const onChange = jest.fn();

	render(<Input placeholder="Input" onChange={onChange} />);

	await userEvent.type(screen.getByRole('textbox'), '1');

	expect(onChange).toHaveBeenCalled();
});
