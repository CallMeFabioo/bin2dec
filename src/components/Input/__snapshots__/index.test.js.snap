// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a input 1`] = `
<div>
  <input
    autocomplete="off"
    class="border rounded shadow w-full px-3 py-2 leading-tight text-gray-700 appearance-none focus:outline-none focus:shadow-outline"
    maxlength="8"
    placeholder="Input"
    type="text"
  />
</div>
`;
